src/entities/friendships.rs

```diff
@@ -289,10 +302,17 @@ impl FriendshipRepositoryImplementation for FriendshipsRepository {
         }
     }
 
-    fn get_executor<'a>(&'a self, transaction: Option<Transaction<'a, Postgres>>) -> Executor<'a> {
+    fn get_executor<'a>(
+        &self,
+        transaction: Option<Transaction<'static, Postgres>>,
+    ) -> Executor<'a> {
         match transaction {
             Some(transaction) => Executor::Transaction(transaction),
-            None => Executor::Pool(DatabaseComponent::get_connection(&self.db_connection)),
+            None => {
+                // Clone because it's cheap and the pool use an Arc internally
```

Comment:
nit: "// Choose to `clone` because it's cheap and the pool uses an Arc internally"